apps/emqx_gateway/src/emqx_gateway_api.erl

```diff
@@ -116,65 +148,38 @@ gateway(post, Request) ->
             return_http_error(404, "Bad gateway name")
     end.
 
-gateway_insta(delete, #{bindings := #{name := Name0}}) ->
-    with_gateway(Name0, fun(GwName, _) ->
-        case emqx_gateway_conf:unload_gateway(GwName) of
-            ok ->
-                {204};
-            {error, Reason} ->
-                emqx_gateway_http:reason2resp(Reason)
+gateway_enable(put, #{bindings := #{name := Name, enable := Enable}}) ->
+    try
+        GwName = binary_to_existing_atom(Name),
+        case emqx_gateway:lookup(GwName) of
+            undefined ->
+                {404, <<"NOT FOUND">>};
+            _Gateway ->
+                case
+                    emqx_gateway_conf:update_gateway(GwName, #{
+                        <<"enable">> => Enable =:= true
```

Comment:
nit: this might be redundant

Reply:
True, since we check it's actually a boolean value in the schema. Didn't think of that.